src/Workspaces/Core/Portable/Shared/Extensions/SyntaxEditorExtensions.cs

```diff
@@ -50,8 +97,9 @@ internal static class SyntaxEditorExtensions
                 editor, document,
                 originalNodes,
                 (syntaxFacts, node) => GetMethodBodySemanticBoundary(syntaxFacts, node),
-                canReplace,
-                updateRoot,
+                (semanticModel, _, node) => canReplace(semanticModel, (TNode)node),
+                (semanticModel, currentRoot, _, node) => updateRoot(semanticModel, currentRoot, (TNode)node),
```

Comment:
the casts here don't make sense to me.  They indicate that the TNode information from this function is being lost in the call to ApplySemanticEditsAsync.  ApplySemanticEditsAsync should be generic on both the 'input type' and the 'type being selected to'.

Reply:
That's because they're wrong :) I mixed up the ordering, and the cast will never make sense. Fixed and removed the cast